apps/prairielearn/src/components/EditQuestionPointsScore.tsx

```diff
@@ -136,3 +144,297 @@ function EditQuestionPointsScoreForm({
+          will be overwritten if the question is answered again by the student.
+          {instanceQuestion.status !== 'unanswered' && (
+            <>
+              {' '}
+              You may also update the score{' '}
+              <a href={manualGradingUrl}>via the manual grading page</a>.
+            </>
+          )}
+        </small>
+      </p>
+      <div class="text-end">
+        <Button disabled={mutation.isPending} variant="secondary" onClick={onCancel}>
+          Cancel
+        </Button>
+        <Button disabled={mutation.isPending} type="submit" variant="primary">
+          {mutation.isPending ? 'Changing...' : 'Change'}
+        </Button>
+      </div>
+    </form>
+  );
+}
+
+interface EditQuestionPointsScoreButtonProps {
+  field: 'points' | 'auto_points' | 'manual_points' | 'score_perc';
+  instanceQuestion: StaffInstanceQuestion;
+  assessmentQuestion: StaffAssessmentQuestion;
+  csrfToken: string;
+  urlPrefix: string;
+  onSuccess?: () => void;
+  onConflict?: (conflictDetailsUrl: string) => void;
+  scrollRef?: React.RefObject<HTMLDivElement> | null;
+}
+
+export function EditQuestionPointsScoreButton({
+  field,
+  instanceQuestion,
+  assessmentQuestion,
+  csrfToken,
+  urlPrefix,
+  onSuccess,
+  onConflict,
+  scrollRef,
+}: EditQuestionPointsScoreButtonProps) {
+  const mutation = useEditQuestionPointsMutation({ csrfToken });
+  const [show, setShow] = useState(false);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  const handleSuccess = () => {
+    setShow(false);
+    onSuccess?.();
+  };
+
+  const handleConflict = (conflictDetailsUrl: string) => {
+    setShow(false);
+    onConflict?.(conflictDetailsUrl);
+  };
+
+  const handleCancel = () => {
+    setShow(false);
+    mutation.reset();
+  };
+
+  // Close popover when table scrolls
+  useEffect(() => {
+    if (!show || !buttonRef.current || !scrollRef) return;
+    const scrollContainer = scrollRef.current;
+    if (!scrollContainer) return;
+
+    const handleScroll = () => {
+      if (!buttonRef.current) return;
+
+      const containerRect = scrollContainer.getBoundingClientRect();
+      const buttonRect = buttonRef.current.getBoundingClientRect();
+
+      // Check if button is outside the visible scroll area
+      const isOutOfView =
+        buttonRect.bottom < containerRect.top ||
+        buttonRect.top > containerRect.bottom ||
+        buttonRect.right < containerRect.left ||
+        buttonRect.left > containerRect.right;
+
+      if (isOutOfView) {
+        setShow(false);
+      }
+    };
+
+    // eslint-disable-next-line react-you-might-not-need-an-effect/no-pass-live-state-to-parent
+    scrollContainer.addEventListener('scroll', handleScroll, { passive: true });
+    return () => {
+      scrollContainer.removeEventListener('scroll', handleScroll);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [show]);
```

Comment:
This works, I might have considered reaching for an `IntersectionObserver` here.